code/core/src/manager/components/sidebar/useFavicon.ts

```diff
@@ -0,0 +1,27 @@
+/* eslint-env browser */
+import { useEffect, useRef } from 'react';
+
+const STATUSES = ['active', 'critical', 'negative', 'positive', 'warning'] as const;
+
+let initialIcon: string | undefined;
+
+const setFavicon = (link: HTMLLinkElement, suffix: string = '') => {
+  initialIcon ??= link.href || `./favicon.svg`;
+
+  const href = initialIcon.replace(/([^/]+)(\.\w+)([?#].*)?$/, `$1${suffix}$2$3`);
+  const img = new Image();
+  img.onload = () => (link.href = href);
+  img.onerror = () => (link.href = initialIcon!);
+  img.src = href;
```

Comment:
We could use something like this here:
```html
<link rel="icon" type="image/svg+xml" href="data:image/svg+xml,[YOUR ENCODED SVG HERE]" />
```

Allowing you to do the masking of an image, and render the icon on top, inline.